clang/test/SemaHLSL/BuiltIns/lit-errors-hlsl202y.hlsl

```diff
@@ -0,0 +1,37 @@
+// RUN: %clang_cc1 -std=hlsl202y -finclude-default-header -x hlsl -triple dxil-pc-shadermodel6.6-library %s -fnative-half-type -emit-llvm-only -disable-llvm-passes -verify
+
+double4 test_double_inputs(double p0, double p1, double p2) {
+  return lit(p0, p1, p2);
+  // expected-error@-1 {{call to deleted function 'lit'}}
```

Comment:
I think these should be ambiguity errors rather than call to deleted function errors, but I'm not sure of an easy way to do this without also allowing implicit vector truncation.

Reply:
I don't think there is an easy way to satisfy both conditions. I will say I'm noticing  a bunch of special cases in DXC where some intrinsics support implicit vector truncation and some don't and its becoming really annoying to apply a consistent rule here. For example dst allows implict vector truncation in DXC so we decided to do the same in clang https://github.com/llvm/llvm-project/pull/133828.